lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3986,34 +3985,70 @@ LogicalResult ConvertAtenOp<AtenClampOp>::matchAndRewrite(
 
   IntegerAttr min_int, max_int;
   FloatAttr min_fp, max_fp;
-  if (op.getMin().getType().isa<Torch::FloatType>()) {
+  if (failed(checkNotNone(rewriter, op, op.getMin()))) {
+    min_int = rewriter.getI64IntegerAttr(std::numeric_limits<int64_t>::min());
+    min_fp = rewriter.getF32FloatAttr(std::numeric_limits<float>::lowest());
+    if (failed(checkNotNone(rewriter, op, op.getMax()))) {
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: either `min` or `max` should be a torch "
+              "constant");
+    } else if (op.getMax().getType().isa<Torch::FloatType>()) {
+      double fp_max;
+      if (!matchPattern(op.getMax(), m_TorchConstantFloat(&fp_max)))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: value `fp_max` should be a torch constant "
+                "float");
+
+      max_int = rewriter.getI64IntegerAttr(static_cast<int64_t>(fp_max));
+      max_fp = rewriter.getF32FloatAttr(static_cast<float>(fp_max));
+    } else {
+      int64_t int_max;
+      if (!matchPattern(op.getMax(), m_TorchConstantInt(&int_max)))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: value `int_max` should be a torch constant "
+                "int");
+
+      max_int = rewriter.getI64IntegerAttr(int_max);
+      max_fp = rewriter.getF32FloatAttr(static_cast<float>(int_max));
+    }
+  } else if (op.getMin().getType().isa<Torch::FloatType>()) {
     double fp_min, fp_max;
     if (!matchPattern(op.getMin(), m_TorchConstantFloat(&fp_min)))
       return rewriter.notifyMatchFailure(
           op, "unimplemented: value `fp_min` should be a torch constant float");
-
-    if (!matchPattern(op.getMax(), m_TorchConstantFloat(&fp_max)))
-      return rewriter.notifyMatchFailure(
-          op, "unimplemented: value `fp_max` should be a torch constant float");
-
     min_int = rewriter.getI64IntegerAttr(static_cast<int64_t>(fp_min));
-    max_int = rewriter.getI64IntegerAttr(static_cast<int64_t>(fp_max));
     min_fp = rewriter.getF32FloatAttr(static_cast<float>(fp_min));
-    max_fp = rewriter.getF32FloatAttr(static_cast<float>(fp_max));
+
+    if (failed(checkNotNone(rewriter, op, op.getMax()))) {
+      max_int = rewriter.getI64IntegerAttr(std::numeric_limits<int64_t>::max());
+      max_fp = rewriter.getF32FloatAttr(std::numeric_limits<float>::max());
+    } else {
+      if (!matchPattern(op.getMax(), m_TorchConstantFloat(&fp_max)))
+        return rewriter.notifyMatchFailure(
+            op,
+            "unimplemented: value `fp_max` should be a torch constant float");
+      max_int = rewriter.getI64IntegerAttr(static_cast<int64_t>(fp_max));
+      max_fp = rewriter.getF32FloatAttr(static_cast<float>(fp_max));
+    }
```

Comment:
We are repeating this code a couple of times. Can we restructure this better to avoid code duplication?

Reply:
Just updated the format. PTAL.